paddle/inference/io.cc

```diff
@@ -73,24 +73,74 @@ void LoadPersistables(framework::Executor& executor,
   delete load_program;
 }
 
-std::unique_ptr<framework::ProgramDesc> Load(framework::Executor& executor,
-                                             framework::Scope& scope,
-                                             const std::string& dirname) {
+// This is called when we have a filen named "param_filename"
+// which has all the persistable variables stored in a
+// combined manner.
+void LoadPersistables(framework::Executor& executor,
+                      framework::Scope& scope,
+                      const std::string& dirname,
+                      const framework::ProgramDesc& main_program,
+                      const std::string& param_filename) {
+  const framework::BlockDesc& global_block = main_program.Block(0);
+
+  framework::ProgramDesc* load_program = new framework::ProgramDesc();
+  framework::BlockDesc* load_block = load_program->MutableBlock(0);
+  std::vector<std::string> paramlist;
+
+  for (auto* var : global_block.AllVars()) {
+    if (IsParameter(var, main_program)) {
+      VLOG(3) << "parameter's name: " << var->Name();
+
+      framework::VarDesc* new_var = load_block->Var(var->Name());
+      new_var->SetShape(var->Shape());
+      new_var->SetDataType(var->GetDataType());
+      new_var->SetType(var->GetType());
+      new_var->SetLoDLevel(var->GetLoDLevel());
+      new_var->SetPersistable(true);
+
+      paramlist.push_back(new_var->Name());
+    }
+  }
+
+  // sort paramlist to have consistent ordering
+  std::sort(paramlist.begin(), paramlist.end());
+
+  // append_op
+  framework::OpDesc* op = load_block->AppendOp();
+  op->SetType("load_combine");
+  op->SetOutput("Out", paramlist);
+  op->SetAttr("file_path", {dirname + "/" + param_filename});
+  op->CheckAttrs();
+  executor.Run(*load_program, &scope, 0, true, true);
+
+  VLOG(3) << "Ran loading successfully";
+  delete load_program;
+}
+
+std::unique_ptr<framework::ProgramDesc> Load(
+    framework::Executor& executor,
+    framework::Scope& scope,
+    const std::string& dirname,
+    const std::string& param_filename) {
```

Comment:
I'd like to keep two `Load()` interfaces as:

```cpp
std::unique_ptr<framework::ProgramDesc> Load(
     framework::Executor& executor,
     framework::Scope& scope,
     const std::string& dirname);

std::unique_ptr<framework::ProgramDesc> Load(
     framework::Executor& executor,
     framework::Scope& scope,
     const std::string& prog_filename,
     const std::string& params_filename);
```

Users are free to rename the file which saving the program. The argument list of `save_inference_model` may be changed in the future. #7995 is merged, we need to test the new `Load`. Please use the new `Load` in the unittest instead.

Reply:
Okay.

Reply:
Done